axum-extra/src/body/async_read_body.rs

```diff
@@ -0,0 +1,96 @@
+use axum::{
+    body::{self, Bytes, HttpBody, StreamBody},
+    http::HeaderMap,
+    response::{IntoResponse, Response},
+    Error,
+};
+use pin_project_lite::pin_project;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tokio::io::AsyncRead;
+use tokio_util::io::ReaderStream;
+
+pin_project! {
+    /// An [`HttpBody`] created from an [`AsyncRead`].
+    ///
+    /// # Example
+    ///
+    /// `AsyncReadBody` can be used to stream the contents of a file:
+    ///
+    /// ```rust
+    /// use axum::{
+    ///     Router,
+    ///     routing::get,
+    ///     http::{StatusCode, header::CONTENT_TYPE},
+    ///     response::{Response, IntoResponse},
+    /// };
+    /// use axum_extra::body::AsyncReadBody;
+    /// use tokio::fs::File;
+    ///
+    /// async fn cargo_toml() -> Result<Response, (StatusCode, String)> {
+    ///     let file = File::open("Cargo.toml")
+    ///         .await
+    ///         .map_err(|err| {
+    ///             (StatusCode::NOT_FOUND, format!("File not found: {}", err))
+    ///         })?;
+    ///
+    ///     let headers = [(CONTENT_TYPE, "text/x-toml")];
```

Comment:
I think we should use `TypedHeader` here, just because it's cool it can be used in responses 😉

Reply:
[`ContentType`](https://docs.rs/headers/latest/headers/struct.ContentType.html#) doesn't have a `toml` constructor and neither does `mime`. So not sure thats possible.

Reply:
Oh 🙁

The `mime` crate should really have a `mime!` macro that does some equivalent of `const { $string_lit.parse::<Mime>().unwrap() }`.

Reply:
Yeah that would be cool